app/controllers/users_controller.rb

```diff
@@ -4,6 +4,6 @@ def index
   end
 
   def show
-    # fetch single user using params[:id]
+    @user = User.find(params[:id]) || 'User not found'
```

Comment:
- Hey, this is making your test not pass as it can't find the user with the id. you can kindly make it only show that user's posts by removing these lines of code. `def show; end`

Reply:
### CREATE POST migration file

- Your migrations fail due to how you've not correctly referenced the foreign key on the user's table as the `author_id` does not exist. kindly update `t.references :author_id, null: false, foreign_key: true` to this 
`t.references :author, foreign_key: { to_table: 'users' }`, then run the migration again.

Once all these issues are addressed, your test will pass without failing.

*After the fixing, it will pass. Issue not fixed? it will fail*
![Screenshot from 2023-04-03 23-37-15](https://user-images.githubusercontent.com/59397159/229641917-40d0e73b-acf1-4afb-99da-598824b3d760.png)

Reply:
issue resolved. thanks

Reply:
From this diagram, we are to use author_id and not author as you are suggesting. @zieeco

Reply:
The test passes yet again as seen in this image.

![test-again](https://user-images.githubusercontent.com/32777408/229643918-65c241da-b236-42ce-bd15-b34052bd366b.png)